streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -450,6 +456,21 @@ public void advanceWallClockTime(final long advanceMs) {
         final V value = valueDeserializer.deserialize(record.topic(), record.value());
         return new ProducerRecord<>(record.topic(), record.partition(), record.timestamp(), key, value);
     }
+    
+    /**
+     * Records sent with this {@link Producer} are streamed to the topology.
+     * <p>
+     * This is useful when testing processes that are also using "normal" kafka producer, typically in
+     * a different thread.<br>
+     * The main difference between using it instead of the {@link #pipeInput(ConsumerRecord)} method is 
+     * that with the producer we both feed the stream and get a topic output (see: {@link #readOutput(String)})
+     * while with {@link #pipeInput(ConsumerRecord)} we only collect output from the topology sinks. 
+     * 
+     * @return the producer.
+     */
+    public final Producer<byte[], byte[]> getProducer() {
```

Comment:
What do you think about calling this `getInputProducer()` or something similar? Depending on their perspective, I think people might mentally associate the producer with the output of the topology instead of the input.

Reply:
I think a Producer is a very well know kind of object in Kafka so in my opinion this is the right name.

Reply:
I agree that the Producer is a well known object in Kafka. But as a user of the TopologyTestDriver, I think it might be unclear to me whether this will give me a handle on a Producer that the TopologyTestDriver will use to produce its *output* as opposed to one that I can use to send it some *input*.